library/asb_get_state.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: asb_get_state
+short_description: Retrieve state from a ConfigMap
+description:
+     - Provides a secure method for allowing Service Bundle developers
+       to access previously set state during a Service Bundle action.
+author:
+    - "Red Hat, Inc."
+options:
+    key:
+        description:
+            - Key of the value that is required
+        required: true
+ env:
+        - Set via the downward API on the APB Pod
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+'''
+
+EXAMPLES = '''
+- name: Fetch some example state
+  asb_get_state:
+    key: "example-key"
+  register: result
+'''
+
+RETURN = '''
+value:
+    description: The value of the provided key
+    returned: success
+    type: string
+    sample: some-value
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(
+        msg="Error attempting to load kubernetes client: {}".format(error))
+
+
+ENV_NAME = "POD_NAME"
+ENV_NAMESPACE = "POD_NAMESPACE"
+CONFIG_PATH = "/etc/apb/state/"
```

Comment:
@PhilipGough  This will be present in the ENV now under BUNDLE_STATE_LOCATION could we update the module to look for that env var?

Reply:
@maleck13 Done